servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultSingleAddressHttpClientBuilder.java

```diff
@@ -552,4 +587,32 @@ public Completable closeAsyncGracefully() {
             return closeable.closeAsyncGracefully();
         }
     }
+
+    private static final class SdStatusCompletable extends Completable {
+        private static final AtomicReferenceFieldUpdater<SdStatusCompletable, CompletableSource.Processor>
+                processorUpdater = newUpdater(SdStatusCompletable.class, CompletableSource.Processor.class,
+                "processor");
+        private volatile CompletableSource.Processor processor = newCompletableProcessor();
+        private boolean seenError;
+
+        @Override
+        protected void handleSubscribe(final Subscriber subscriber) {
+            processor.subscribe(subscriber);
+        }
+
+        void nextError(final Throwable t) {
+            final CompletableSource.Processor newProcessor = newCompletableProcessor();
+            newProcessor.onError(t);
+            final CompletableSource.Processor oldProcessor = processorUpdater.getAndSet(this, newProcessor);
+            oldProcessor.onError(t);
```

Comment:
nit: inline `oldProcessor` 
```suggestion
            processorUpdater.getAndSet(this, newProcessor).onError(t);
```